metpy/gridding/triangles.py

```diff
@@ -0,0 +1,376 @@
+    members: dictionary
+        List of simplex codes for natural neighbor
+        triangles in 'tri' for each grid cell.
+    triangle_info: dictionary
+        Circumcenter and radius information for each
+        triangle in 'tri'.
+    """
+
+    tree = cKDTree(grid_points)
+
+    in_triangulation = tri.find_simplex(tree.data) >= 0
+
+    triangle_info = dict()
+
+    members = dict((key, []) for key in range(len(tree.data)))
+
+    for i in range(len(tri.simplices)):
+
+        ps = tri.points[tri.simplices[i]]
+
+        cc = circumcenter(*ps)
+        r = circumcircle_radius(*ps)
+
+        triangle_info[i] = {'cc': cc, 'r': r}
+
+        qualifiers = tree.query_ball_point(cc, r)
+
+        for qualifier in qualifiers:
+            if in_triangulation[qualifier]:
+                members[qualifier].append(i)
+
+    return members, triangle_info
+
+
+@exporter.export
+def find_nn_triangles_point(tri, cur_tri, point):
+    r"""Returns the natural neighbors of a triangle
+    containing a point based on the provided
+    Delaunay Triangulation.
+    Parameters
+    ----------
+    tri: Object
+        A Delaunay Triangulation
+    cur_tri: int
+        Simplex code for Delaunay Triangulation lookup of
+        a given triangle that contains 'position'.
+    point: (x, y)
+        Coordinates used to calculate distances to
+        simplexes in 'tri'.
+    Returns
+    --------
+    nn: (N, ) array
+        List of simplex codes for natural neighbor
+        triangles in 'tri'.
+    """
+
+    nn = []
+
+    candidates = set(tri.neighbors[cur_tri])
+
+    # find the union of the two sets
+    candidates |= set(tri.neighbors[tri.neighbors[cur_tri]].flat)
+
+    # remove instances of the "no neighbors" code
+    candidates.discard(-1)
+
+    for neighbor in candidates:
+
+        triangle = tri.points[tri.simplices[neighbor]]
+        cur_x, cur_y = circumcenter(triangle[0], triangle[1], triangle[2])
+        r = circumcircle_radius_2(triangle[0], triangle[1], triangle[2])
+
+        if dist_2(point[0], point[1], cur_x, cur_y) < r:
+
+            nn.append(neighbor)
+
+    return nn
+
+
+@exporter.export
+def find_local_boundary(tri, triangles):
+    r"""Finds and returns the outside edges of a collection
+    of natural neighbor triangles.  There is no guarantee
+    that this boundary is convex, so ConvexHull is not
+    sufficient in some situations.
+
+    Parameters
+    ----------
+    tri: Object
+        A Delaunay Triangulation
+    triangles: (N, ) array
+        List of natural neighbor triangles.
+
+    Returns
+    --------
+    edges: (2, N) ndarray
+        List of vertex codes that form outer edges of
+        a group of natural neighbor triangles.
+    """
+
+    edges = []
+
+    for triangle in triangles:
+
+        for i in range(3):
+
+            pt1 = tri.simplices[triangle][i]
+            pt2 = tri.simplices[triangle][(i + 1) % 3]
+
+            if (pt1, pt2) in edges:
+                edges.remove((pt1, pt2))
```

Comment:
According to code coverage, this line is never executed. Is there some algorithmic reason that the points will always be `(pt2, pt1)` in `edges`? If so, we could simplify here.